datasource/opennms-kafka/src/main/java/org/opennms/oce/datasource/opennms/OpennmsDatasource.java

```diff
@@ -492,38 +508,60 @@ public void unregisterHandler(InventoryHandler handler) {
                 situations.add(OpennmsMapper.toSituation(entry.value));
             });
         } catch (InterruptedException e) {
+            Thread.currentThread().interrupt();
             throw new RuntimeException(e);
+        } catch (TimeoutException e) {
+            LOG.info("{}, returning empty list", e.getMessage());
         }
         return situations;
     }
 
-    private ReadOnlyKeyValueStore<String, OpennmsModelProtos.Alarm> waitUntilSituationStoreIsQueryable() throws InterruptedException {
+    private ReadOnlyKeyValueStore<String, OpennmsModelProtos.Alarm> waitUntilSituationStoreIsQueryable()
+            throws InterruptedException, TimeoutException {
         return waitUntilStoreIsQueryable(SITUATION_STORE);
     }
 
-    private ReadOnlyKeyValueStore<String, OpennmsModelProtos.Alarm> waitUntilAlarmStoreIsQueryable() throws InterruptedException {
+    private ReadOnlyKeyValueStore<String, OpennmsModelProtos.Alarm> waitUntilAlarmStoreIsQueryable()
+            throws InterruptedException, TimeoutException {
         return waitUntilStoreIsQueryable(ALARM_STORE);
     }
 
-    private ReadOnlyKeyValueStore<String, InventoryModelProtos.InventoryObjects> waitUntilInventoryStoreIsQueryable() throws InterruptedException {
+    private ReadOnlyKeyValueStore<String, InventoryModelProtos.InventoryObjects> waitUntilInventoryStoreIsQueryable()
+            throws InterruptedException, TimeoutException {
         return waitUntilStoreIsQueryable(INVENTORY_STORE);
     }
 
-    private ReadOnlyKeyValueStore<String, FeedbackModelProtos.AlarmFeedbacks> waitUntilAlarmFeedbackStoreIsQueryable() throws InterruptedException {
+    private ReadOnlyKeyValueStore<String, FeedbackModelProtos.AlarmFeedbacks> waitUntilAlarmFeedbackStoreIsQueryable()
+            throws InterruptedException, TimeoutException {
         return waitUntilStoreIsQueryable(ALARM_FEEDBACK_STORE);
     }
 
-    private <K,V> ReadOnlyKeyValueStore<K, V> waitUntilStoreIsQueryable(String storeName) throws InterruptedException {
+    private <K, V> ReadOnlyKeyValueStore<K, V> waitUntilStoreIsQueryable(String storeName)
+            throws InterruptedException, TimeoutException {
         if (streams == null) {
             throw new IllegalStateException("Datasource must be started first.");
         }
-        while (true) {
+
+        Stopwatch stopwatch = Stopwatch.createStarted();
+
+        while (stopwatch.elapsed(TimeUnit.SECONDS) <= 20) {
```

Comment:
Can we make the value of `20` configurable?

Reply:
Done, sanity tested defaults with the e2e test to make sure nothing broke afterwards.